lib/private/Files/Storage/DAV.php

```
@@ -274,11 +269,15 @@ protected function propfind($path) {
 					]
 				);
 				$this->statCache->set($path, $response);
-			} catch (NotFound $e) {
-				// remember that this path did not exist
-				$this->statCache->clear($path . '/');
-				$this->statCache->set($path, false);
-				throw $e;
+			} catch (ClientHttpException $e) {
```

It's a bit messy...

Basically Sabre's propfind itself only ever throws `ClientHttpException` or `ClientException`, never `NotFound` (that one belongs to sabre server, not sabre client btw).

But here we do want to throw a known exception for the consumers of our own `$this->propfind`, so we throw `NotFound` and they all catch it.

Maybe it would be cleaner to simply rethrow `ClientHttpException` as is to avoid confusion ?